### PR TITLE
[new release] biotk (0.2.0)

### PR DESCRIPTION
**Pull request by pveber**
Bioinformatics toolkit

- Project page: <a href="https://github.com/pveber/biotk/">https://github.com/pveber/biotk/</a>

##### CHANGES:

- changed license to CeCILL-C, for LGPL compatibility
- dropped dependency on biocaml, missing functionalities have been
  imported and adapted
- croquis: plotting functions, several enhancements and bug fixes
- new Amino_acid module
- new Alphabet module
- new Meme module
- Compatible with core v0.15.0
- Improvement in Dataframe parsing

### CODE REVIEW

**Comment by dinosaure** [comment]
`Seq.concat` (used into the codebase) appears only since 4.13. Can you up the lower bound about OCaml to 4.13?

**Comment by dinosaure** [comment]
And a constraint on `gsl` should help the CI to compile your project - currently, it fails due to old version of it.

**Comment by pveber** [comment]
Thanks @dinosaure! I've fixed the bound on `ocaml`. However I believe the build fail on `gsl` is related to a [bug in this library](https://github.com/mmottl/gsl-ocaml/pull/31). A fix was proposed by @rgrinberg and merged by @mmottl but it hasn't made it yet in a release. So adding a constraint on `gsl` should not help here.

File: packages/biotk/biotk.0.2.0/opam [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
+  ]
+]
+dev-repo: "git+https://github.com/pveber/biotk.git"
```
Comment (mseri) [inline packages/biotk/biotk.0.2.0/opam:52]:
```suggestion
available: arch != "arm32" & arch != "x86_32"
dev-repo: "git+https://github.com/pveber/biotk.git"
```

Due to
```

#=== ERROR while compiling biotk.0.2.0 ========================================#
# context              2.2.0~alpha~dev | linux/x86_32 | ocaml-base-compiler.4.14.1 | pinned(https://github.com/pveber/biotk/releases/download/v0.2.0/biotk-0.2.0.tbz)
# path                 ~/.opam/4.14/.opam-switch/build/biotk.0.2.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p biotk -j 71 @install
# exit-code            1
# env-file             ~/.opam/log/biotk-7-9c5ab1.env
# output-file          ~/.opam/log/biotk-7-9c5ab1.out
### output ###
# (cd _build/default/lib/croquis && /home/opam/.opam/4.14/bin/ocaml-crunch --mode plain ../../fonts --output font_data.ml)
# Generating font_data.ml
# Skipping generation of .mli
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlopt.opt -g -w -24 -o .ppx/b384f542a164a0205c2727109e0fadf9/ppx.exe /home/opam/.opam/4.14/lib/base/base_internalhash_types/base_internalhash_types.cmxa -I /home/opam/.opam/4.14/lib/base/base_internalhash_types /home/opam/.opam/4.14/lib/base/caml/caml.cmxa /home/opam/.opam/4.14/lib/sexplib0/sexplib0.cmxa /home/opam/.opam/4.14/lib/base/shadow_stdlib/shadow_stdlib.cmxa /home/opam/.opam/4.14/lib/base/base.cmxa -I /home/opam/.opam/4.14/lib/base /home/opam/.opam/4.14/lib/ocaml/compiler-libs/ocamlcommon.cmxa /home/opam/.opam/4.14/lib/ocaml-compiler-libs/common/ocaml_common.cmxa /home/opam/.opam/4.14/lib/ppxlib/astlib/astlib.cmxa /home/opam/.opam/4.14/lib/stdlib-shims/stdlib_shims.cmxa /home/opam/.opam/4.14/lib/ppxlib/ast/ppxlib_ast.cmxa /home/opam/.opam/4.14/lib/ocaml-compiler-libs/shadow/ocaml_shadow.cmxa /home/opam/.opam/4.14/lib/ppxlib/print_diff/ppxlib_print_diff.cmxa /home/opam/.opam/4.14/lib/ppx_derivers/ppx_derivers.cmxa /home/opam/.opam/4.14/lib/ppxlib/traverse_builtins/ppxlib_traverse_builtins.cmxa /home/opam/.opam/4.14/lib/ppxlib/stdppx/stdppx.cmxa /home/opam/.opam/4.14/lib/ppxlib/ppxlib.cmxa /home/opam/.opam/4.14/lib/ppx_compare/expander/ppx_compare_expander.cmxa /home/opam/.opam/4.14/lib/ppx_compare/ppx_compare.cmxa /home/opam/.opam/4.14/lib/ppx_conv_func/ppx_conv_func.cmxa /home/opam/.opam/4.14/lib/ppx_fields_conv/ppx_fields_conv.cmxa /home/opam/.opam/4.14/lib/ppx_csv_conv/ppx_csv_conv.cmxa /home/opam/.opam/4.14/lib/result/result.cmxa /home/opam/.opam/4.14/lib/ppx_deriving/api/ppx_deriving_api.cmxa /home/opam/.opam/4.14/lib/ppx_deriving/show/ppx_deriving_show.cmxa /home/opam/.opam/4.14/lib/ppx_expect/common/expect_test_common.cmxa /home/opam/.opam/4.14/lib/ppx_expect/payload/ppx_expect_payload.cmxa /home/opam/.opam/4.14/lib/ppx_inline_test/libname/ppx_inline_test_libname.cmxa /home/opam/.opam/4.14/lib/ppx_inline_test/ppx_inline_test.cmxa /home/opam/.opam/4.14/lib/ppx_here/expander/ppx_here_expander.cmxa /home/opam/.opam/4.14/lib/ppx_expect/ppx_expect.cmxa /home/opam/.opam/4.14/lib/ppx_let/expander/ppx_let_expander.cmxa /home/opam/.opam/4.14/lib/ppx_let/ppx_let.cmxa /home/opam/.opam/4.14/lib/ppxlib/metaquot_lifters/ppxlib_metaquot_lifters.cmxa /home/opam/.opam/4.14/lib/ppx_sexp_conv/expander/ppx_sexp_conv_expander.cmxa /home/opam/.opam/4.14/lib/ppx_sexp_conv/ppx_sexp_conv.cmxa .ppx/b384f542a164a0205c2727109e0fadf9/dune__exe___ppx.cmx)
# /usr/bin/ld: /home/opam/.opam/4.14/lib/ppx_sexp_conv/ppx_sexp_conv.a(ppx_sexp_conv.o): warning: relocation in read-only section `.text'
# /usr/bin/ld: warning: creating DT_TEXTREL in a PIE
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w -40 -short-paths -g -bin-annot -I lib/.biotk.objs/byte -I /home/opam/.opam/4.14/lib/angstrom -I /home/opam/.opam/4.14/lib/angstrom-unix -I /home/opam/.opam/4.14/lib/base -I /home/opam/.opam/4.14/lib/base/base_internalhash_types -I /home/opam/.opam/4.14/lib/base/caml -I /home/opam/.opam/4.14/lib/base/md5 -I /home/opam/.opam/4.14/lib/base/shadow_stdlib -I /home/opam/.opam/4.14/lib/base_bigstring -I /home/opam/.opam/4.14/lib/base_quickcheck -I /home/opam/.opam/4.14/lib/base_quickcheck/ppx_quickcheck/runtime -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/bin_prot -I /home/opam/.opam/4.14/lib/bin_prot/shape -I /home/opam/.opam/4.14/lib/binning -I /home/opam/.opam/4.14/lib/core -I /home/opam/.opam/4.14/lib/core/base_for_tests -I /home/opam/.opam/4.14/lib/core/validate -I /home/opam/.opam/4.14/lib/core_kernel -I /home/opam/.opam/4.14/lib/core_kernel/binary_packing -I /home/opam/.opam/4.14/lib/core_kernel/caml_threads -I /home/opam/.opam/4.14/lib/core_kernel/caml_unix -I /home/opam/.opam/4.14/lib/core_kernel/flags -I /home/opam/.opam/4.14/lib/core_unix -I /home/opam/.opam/4.14/lib/core_unix/error_checking_mutex -I /home/opam/.opam/4.14/lib/core_unix/signal_unix -I /home/opam/.opam/4.14/lib/csvfields -I /home/opam/.opam/4.14/lib/csvfields/csvlib -I /home/opam/.opam/4.14/lib/csvfields/xml-light -I /home/opam/.opam/4.14/lib/fieldslib -I /home/opam/.opam/4.14/lib/fmt -I /home/opam/.opam/4.14/lib/gg -I /home/opam/.opam/4.14/lib/gsl -I /home/opam/.opam/4.14/lib/int_repr -I /home/opam/.opam/4.14/lib/jane-street-headers -I /home/opam/.opam/4.14/lib/ocaml/threads -I /home/opam/.opam/4.14/lib/otfm -I /home/opam/.opam/4.14/lib/parsexp -I /home/opam/.opam/4.14/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_deriving/runtime -I /home/opam/.opam/4.14/lib/ppx_enumerate/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_expect/collector -I /home/opam/.opam/4.14/lib/ppx_expect/common -I /home/opam/.opam/4.14/lib/ppx_expect/config -I /home/opam/.opam/4.14/lib/ppx_expect/config_types -I /home/opam/.opam/4.14/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_here/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_inline_test/config -I /home/opam/.opam/4.14/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_log/types -I /home/opam/.opam/4.14/lib/ppx_module_timer/runtime -I /home/opam/.opam/4.14/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.14/lib/re -I /home/opam/.opam/4.14/lib/result -I /home/opam/.opam/4.14/lib/rresult -I /home/opam/.opam/4.14/lib/seq -I /home/opam/.opam/4.14/lib/sexplib -I /home/opam/.opam/4.14/lib/sexplib/unix -I /home/opam/.opam/4.14/lib/sexplib0 -I /home/opam/.opam/4.14/lib/spawn -I /home/opam/.opam/4.14/lib/splittable_random -I /home/opam/.opam/4.14/lib/stdio -I /home/opam/.opam/4.14/lib/stringext -I /home/opam/.opam/4.14/lib/time_now -I /home/opam/.opam/4.14/lib/typerep -I /home/opam/.opam/4.14/lib/tyxml -I /home/opam/.opam/4.14/lib/tyxml/functor -I /home/opam/.opam/4.14/lib/uri -I /home/opam/.opam/4.14/lib/uutf -I /home/opam/.opam/4.14/lib/variantslib -I /home/opam/.opam/4.14/lib/vg -I /home/opam/.opam/4.14/lib/zip -I lib/croquis/.biotk_croquis.objs/public_cmi -intf-suffix .ml -no-alias-deps -open Biotk -o lib/.biotk.objs/byte/biotk__Sam.cmo -c -impl lib/sam.pp.ml)
# File "lib/sam.ml", line 218, characters 34-44:
# 218 |   (if (1 <= length) && (length <= 2147483647) then
#                                         ^^^^^^^^^^
# Error: Integer literal exceeds the range of representable integers of type int
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w -40 -short-paths -g -bin-annot -I lib/.biotk.objs/byte -I /home/opam/.opam/4.14/lib/angstrom -I /home/opam/.opam/4.14/lib/angstrom-unix -I /home/opam/.opam/4.14/lib/base -I /home/opam/.opam/4.14/lib/base/base_internalhash_types -I /home/opam/.opam/4.14/lib/base/caml -I /home/opam/.opam/4.14/lib/base/md5 -I /home/opam/.opam/4.14/lib/base/shadow_stdlib -I /home/opam/.opam/4.14/lib/base_bigstring -I /home/opam/.opam/4.14/lib/base_quickcheck -I /home/opam/.opam/4.14/lib/base_quickcheck/ppx_quickcheck/runtime -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/bin_prot -I /home/opam/.opam/4.14/lib/bin_prot/shape -I /home/opam/.opam/4.14/lib/binning -I /home/opam/.opam/4.14/lib/core -I /home/opam/.opam/4.14/lib/core/base_for_tests -I /home/opam/.opam/4.14/lib/core/validate -I /home/opam/.opam/4.14/lib/core_kernel -I /home/opam/.opam/4.14/lib/core_kernel/binary_packing -I /home/opam/.opam/4.14/lib/core_kernel/caml_threads -I /home/opam/.opam/4.14/lib/core_kernel/caml_unix -I /home/opam/.opam/4.14/lib/core_kernel/flags -I /home/opam/.opam/4.14/lib/core_unix -I /home/opam/.opam/4.14/lib/core_unix/error_checking_mutex -I /home/opam/.opam/4.14/lib/core_unix/signal_unix -I /home/opam/.opam/4.14/lib/csvfields -I /home/opam/.opam/4.14/lib/csvfields/csvlib -I /home/opam/.opam/4.14/lib/csvfields/xml-light -I /home/opam/.opam/4.14/lib/fieldslib -I /home/opam/.opam/4.14/lib/fmt -I /home/opam/.opam/4.14/lib/gg -I /home/opam/.opam/4.14/lib/gsl -I /home/opam/.opam/4.14/lib/int_repr -I /home/opam/.opam/4.14/lib/jane-street-headers -I /home/opam/.opam/4.14/lib/ocaml/threads -I /home/opam/.opam/4.14/lib/otfm -I /home/opam/.opam/4.14/lib/parsexp -I /home/opam/.opam/4.14/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_deriving/runtime -I /home/opam/.opam/4.14/lib/ppx_enumerate/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_expect/collector -I /home/opam/.opam/4.14/lib/ppx_expect/common -I /home/opam/.opam/4.14/lib/ppx_expect/config -I /home/opam/.opam/4.14/lib/ppx_expect/config_types -I /home/opam/.opam/4.14/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_here/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_inline_test/config -I /home/opam/.opam/4.14/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_log/types -I /home/opam/.opam/4.14/lib/ppx_module_timer/runtime -I /home/opam/.opam/4.14/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.14/lib/re -I /home/opam/.opam/4.14/lib/result -I /home/opam/.opam/4.14/lib/rresult -I /home/opam/.opam/4.14/lib/seq -I /home/opam/.opam/4.14/lib/sexplib -I /home/opam/.opam/4.14/lib/sexplib/unix -I /home/opam/.opam/4.14/lib/sexplib0 -I /home/opam/.opam/4.14/lib/spawn -I /home/opam/.opam/4.14/lib/splittable_random -I /home/opam/.opam/4.14/lib/stdio -I /home/opam/.opam/4.14/lib/stringext -I /home/opam/.opam/4.14/lib/time_now -I /home/opam/.opam/4.14/lib/typerep -I /home/opam/.opam/4.14/lib/tyxml -I /home/opam/.opam/4.14/lib/tyxml/functor -I /home/opam/.opam/4.14/lib/uri -I /home/opam/.opam/4.14/lib/uutf -I /home/opam/.opam/4.14/lib/variantslib -I /home/opam/.opam/4.14/lib/vg -I /home/opam/.opam/4.14/lib/zip -I lib/croquis/.biotk_croquis.objs/public_cmi -intf-suffix .ml -no-alias-deps -open Biotk -o lib/.biotk.objs/byte/biotk__Bam.cmo -c -impl lib/bam.pp.ml)
# File "lib/bam.ml", line 662, characters 13-23:
# 662 |       | Some 2147483647 (* POS in BAM is 0-based *)
#                    ^^^^^^^^^^
# Error: Integer literal exceeds the range of representable integers of type int
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlopt.opt -w -40 -short-paths -g -I lib/.biotk.objs/byte -I lib/.biotk.objs/native -I /home/opam/.opam/4.14/lib/angstrom -I /home/opam/.opam/4.14/lib/angstrom-unix -I /home/opam/.opam/4.14/lib/base -I /home/opam/.opam/4.14/lib/base/base_internalhash_types -I /home/opam/.opam/4.14/lib/base/caml -I /home/opam/.opam/4.14/lib/base/md5 -I /home/opam/.opam/4.14/lib/base/shadow_stdlib -I /home/opam/.opam/4.14/lib/base_bigstring -I /home/opam/.opam/4.14/lib/base_quickcheck -I /home/opam/.opam/4.14/lib/base_quickcheck/ppx_quickcheck/runtime -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/bin_prot -I /home/opam/.opam/4.14/lib/bin_prot/shape -I /home/opam/.opam/4.14/lib/binning -I /home/opam/.opam/4.14/lib/core -I /home/opam/.opam/4.14/lib/core/base_for_tests -I /home/opam/.opam/4.14/lib/core/validate -I /home/opam/.opam/4.14/lib/core_kernel -I /home/opam/.opam/4.14/lib/core_kernel/binary_packing -I /home/opam/.opam/4.14/lib/core_kernel/caml_threads -I /home/opam/.opam/4.14/lib/core_kernel/caml_unix -I /home/opam/.opam/4.14/lib/core_kernel/flags -I /home/opam/.opam/4.14/lib/core_unix -I /home/opam/.opam/4.14/lib/core_unix/error_checking_mutex -I /home/opam/.opam/4.14/lib/core_unix/signal_unix -I /home/opam/.opam/4.14/lib/csvfields -I /home/opam/.opam/4.14/lib/csvfields/csvlib -I /home/opam/.opam/4.14/lib/csvfields/xml-light -I /home/opam/.opam/4.14/lib/fieldslib -I /home/opam/.opam/4.14/lib/fmt -I /home/opam/.opam/4.14/lib/gg -I /home/opam/.opam/4.14/lib/gsl -I /home/opam/.opam/4.14/lib/int_repr -I /home/opam/.opam/4.14/lib/jane-street-headers -I /home/opam/.opam/4.14/lib/ocaml/threads -I /home/opam/.opam/4.14/lib/otfm -I /home/opam/.opam/4.14/lib/parsexp -I /home/opam/.opam/4.14/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_deriving/runtime -I /home/opam/.opam/4.14/lib/ppx_enumerate/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_expect/collector -I /home/opam/.opam/4.14/lib/ppx_expect/common -I /home/opam/.opam/4.14/lib/ppx_expect/config -I /home/opam/.opam/4.14/lib/ppx_expect/config_types -I /home/opam/.opam/4.14/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_here/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_inline_test/config -I /home/opam/.opam/4.14/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_log/types -I /home/opam/.opam/4.14/lib/ppx_module_timer/runtime -I /home/opam/.opam/4.14/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.14/lib/re -I /home/opam/.opam/4.14/lib/result -I /home/opam/.opam/4.14/lib/rresult -I /home/opam/.opam/4.14/lib/seq -I /home/opam/.opam/4.14/lib/sexplib -I /home/opam/.opam/4.14/lib/sexplib/unix -I /home/opam/.opam/4.14/lib/sexplib0 -I /home/opam/.opam/4.14/lib/spawn -I /home/opam/.opam/4.14/lib/splittable_random -I /home/opam/.opam/4.14/lib/stdio -I /home/opam/.opam/4.14/lib/stringext -I /home/opam/.opam/4.14/lib/time_now -I /home/opam/.opam/4.14/lib/typerep -I /home/opam/.opam/4.14/lib/tyxml -I /home/opam/.opam/4.14/lib/tyxml/functor -I /home/opam/.opam/4.14/lib/uri -I /home/opam/.opam/4.14/lib/uutf -I /home/opam/.opam/4.14/lib/variantslib -I /home/opam/.opam/4.14/lib/vg -I /home/opam/.opam/4.14/lib/zip -I lib/croquis/.biotk_croquis.objs/native -I lib/croquis/.biotk_croquis.objs/public_cmi -intf-suffix .ml -no-alias-deps -open Biotk -o lib/.biotk.objs/native/biotk__Sam.cmx -c -impl lib/sam.pp.ml)
# File "lib/sam.ml", line 218, characters 34-44:
# 218 |   (if (1 <= length) && (length <= 2147483647) then
#                                         ^^^^^^^^^^
# Error: Integer literal exceeds the range of representable integers of type int
```

Comment (pveber) [reply packages/biotk/biotk.0.2.0/opam:53]: Thanks @mseri! I had missed this one at the time, but it's added now. Thanks again!

**Pull request merged**